subprojects/packagefiles/win-iconv/meson.build

```diff
@@ -0,0 +1,66 @@
+project(
+  'win-iconv',
+  'c',
+  version: '0.0.8',
+  meson_version: '>=1.1.0',
+)
+
+if host_machine.system() != 'windows'
+  error('win-iconv is only supported on windows, and not supported on '
+    + host_machine.system())
+endif
+
+cc = meson.get_compiler('c')
+
+c_args = []
+
+if cc.get_argument_syntax() == 'msvc'
+  c_args += '/D_CRT_SECURE_NO_WARNINGS'
```

Comment:
Meson will auto-convert `-D`-style defines.

Reply:
In other wraps, there's a trick to use the /D syntax with get_supported_arguments to apply this define only on MSVC compilers. _CRT_SECURE_NO_WARNING is only valid there.

Reply:
yeah.

```
/usr/x86_64-w64-mingw32/include$ grep _CRT_SECURE_NO_WARNINGS *.h
_mingw_mac.h:  "This function or variable may be unsafe, use _CRT_SECURE_NO_WARNINGS to disable deprecation"
_mingw_mac.h:#if !defined (_CRT_SECURE_NO_WARNINGS) || (_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES == 0)
```

Reply:
MinGW headers also use `_CRT_SECURE_NO_WARNINGS` to silence warnings, and in that case the compiler and the argument syntax are GCC.

Reply:
hrm looking at it more carefully, you might be right. Interestingly even though there's a define to print a deprecation message, I haven't seen it in practice. I wonder why...

Reply:
I changed it to `-D` as it can be also used under MSYS2, see the other comment https://github.com/mesonbuild/wrapdb/pull/2048#discussion_r2051030852